src/main/kotlin/com/zaxxer/ping/impl/util/WaitingTargetCollection.kt

```diff
@@ -2,44 +2,46 @@ package com.zaxxer.ping.impl.util
 
 import com.zaxxer.ping.PingTarget
 import com.carrotsearch.hppc.ShortObjectHashMap
-import java.util.concurrent.PriorityBlockingQueue
-
-private const val INITIAL_CAPACITY = 16
+import java.util.TreeSet
 
+/**
+ * This class is not thread safe.
+ */
 class WaitingTargetCollection {
     private val waitingTargetMap = ShortObjectHashMap<PingTarget>()
-    private val targetTimeoutQueue = PriorityBlockingQueue<PingTarget>(INITIAL_CAPACITY)
+    /**
+     * Values are ordered by timeout and sequence.
+     */
+    private val targetTimeoutQueue = TreeSet<PingTarget>()
```

Comment:
What is the purpose of the change from ``PriorityBlockingQueue`` to ``TreeSet``? ``PriorityBlockingQueue.poll()`` is an *O(1)* operation, whereas any removal from a ``TreeSet`` including ``removeFirst()`` is an *O(log n)* operation.